cmake/CMakeLists.txt

```diff
@@ -233,6 +233,15 @@ if(onnxruntime_BUILD_SHARED_LIB OR onnxruntime_ENABLE_PYTHON)
   endif()
 endif()
 
+# check endian-ness and add appropriate compile definition
+include(TestBigEndian)
```

Comment:
What if it is cross-compiling?

Reply:
I think this is what CMake does, it compiles a test program. Please, verify that from CMake build logs. It should clearly show it there.

---
In reply to: [336639449](https://github.com/microsoft/onnxruntime/pull/2181#discussion_r336639449) [](ancestors = 336639449)

Reply:
Yes, the logs show that it does compile a test program. CMake searches the binary for some constant strings that should be present only in the big/little-endian case.

Check here for details:
https://github.com/Kitware/CMake/blob/master/Modules/TestBigEndian.cmake
https://github.com/Kitware/CMake/blob/master/Modules/TestEndianess.c.in

Reply:
But that requires everyone who consumes our C++ API to do the same thing in their cmake project. I'm wondering if it is worth to bring in such a big change.

Reply:
Ok, I removed the usage of CMake's TestBigEndian module. However, I'm more confident in CMake to get that right than myself, so please take a look :).